File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem15/Problem15.java
public class Problem15 {

    public static void chessBoard(int[] a, int num){
        if (num == 8) {
            for (int k = 0; k < 8; k++) {
                System.out.print(a[k] + ((k == 7) ? "" : " | "));
            }
            System.out.println();
        } else {
            for (int i = 0; i < 8; i++) {
                boolean isTrue = true;
                for (int j = 0; j < num; j++) {
                    if (a[j] == i || i - num == a[j] - j || i + num == a[j] + j) {
                        isTrue = false;
                    }
                }
                if (isTrue) {
                    a[num] = i;
                    chessBoard(a, num + 1);
                }
            }
        }
        
    }

    public static void main(String[] args) {
        int[] a = new int[8];
        int num = 0;

        chessBoard(a, num);
    }
}
    

File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_15/problem_15.py
def chess_board(a, num):
    if num == 8:
        for i in range(0, 8):
            print(f"{a[i]}", end=" / ")
        print()
    else:
        for i in range(0, 8):
            is_true = True
            for j in range(0, num):
                if (i == a[j]) or (i - num == a[j] - j) or (i + num == a[j] + j):
                    is_true = False
            if is_true:
                a[num] = i
                chess_board(a, num + 1)
                    
            
def main():
    a = [0, 0, 0, 0, 0, 0, 0, 0]
    num = 0

    chess_board(a, num)

if __name__ == "__main__":
    main()